### DISCUSSION TITLE
Prisma Operational Telemetry

### DISCUSSION THREAD

**Discussion by jmickey** [prisma/prisma#6828] [Q&A] [answered]
I'm running Prisma in a production application that is more write than read heavy. Right now we're seeing about 40-60 write ops/s.

A very small percentage of these writes take a significant amount time more than the rest. 50p latency is about 140ms, 95p at a little over 900ms, and 99p around 4.5s. 

Based on my current troubleshooting there doesn't seem to be anything special in particular about the data shape or size of the operations that take significantly longer. So my question(s) are basically as follows:

- Does Prisma provide a way to see certain information at the time of an operation - e.g. connection pool saturation, queue time, delays in achieving DB locks, etc.
- As with the above question, does Prisma provide a way to expose internal metrics (e.g. conn pool size, queue time) via a prometheus endpoint? We already use prometheus in our environment, so it would be awesome if Prisma was able to hook into the prom-client and expose internal information.
- Are there any recommended strategies for profiling applications using prisma in production? What tools have folks had success with?

Thanks in advance for your time!

**Comment by janpio** [comment, marked as answer]
Prisma does not give good guidance here yet. Do you maybe want to open a feature request issue and describe in more details what you would expect to see here? We are very much looking for input here.

Right now your best solution will probably be to watch your database directly, and see if that already gives you the answers you are looking for.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
